File: spring-web-java/src/main/java/spring/web/java/presentation/dto/response/ItemResponse.java
package spring.web.java.presentation.dto.response;

import java.time.Instant;

import spring.web.java.domain.model.entity.Item;

public record ItemResponse(
	Long id,
	String name,
	String description,
	int price,
	int quantity,
	Instant createdAt
) {
	public ItemResponse(Item item) {
		this(
			item.getId(),
			item.getName(),
			item.getDescription(),
			item.getPrice(),
			item.getQuantity(),
			item.getCreatedAt()
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/response/ItemResponse.kt
package spring.web.kotlin.presentation.dto.response

import spring.web.kotlin.domain.model.entity.Item
import java.time.Instant

data class ItemResponse(
    val id: Long,
    val name: String,
    val description: String,
    val price: Int,
    val quantity: Int,
    val createdAt: Instant,
) {
    constructor(item: Item) : this(
        id = requireNotNull(item.id),
        name = item.name,
        description = item.description,
        price = item.price,
        quantity = item.quantity,
        createdAt = item.createdAt,
    )
}
